### PR TITLE
Wagtail 5.2 Support

### PR DESCRIPTION
**Pull request by jams2**
Adds support for Wagtail 5.2. Between 5.1 and 5.2 the `wagtailsnippets/snippets/index.html` template diverges in a way that maintaining compatibility for both versions would be a burden, so I propose releasing the next `wagtail-airtable` version without support for Wagtail < 5.2 (suggested by @zerolab).

Changes:
- use `wagtailsnippets/snippets/index.html` instead of `wagtailsnippets/snippets/type_index.html` (removed)
- update long-running buttons to use new styles and stimulus.js declarations
- add a redirect back to the snippet list view when you're importing airtable data from the snippet list view - I think this is a better user experience than being redirected to the airtable import view.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -15,27 +15,26 @@ classifiers=[
...
+    'Framework :: Wagtail :: 5',
 ]
 packages = [
     { include = "wagtail_airtable" },
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.7,<4.0"
-wagtail = ">=4.1"
+python = ">=3.8,<3.13"
```
Comment (zerolab) [inline pyproject.toml:33]: question: why not leave simply as `>=3.8`?

File: wagtail_airtable/templates/wagtail_airtable/airtable_import_listing.html [hunk 2]
```diff
@@ -1,32 +1,35 @@
 {% extends "wagtailadmin/base.html" %}
-{% load wagtailadmin_tags %}
+{% load i18n wagtailadmin_tags %}
```
Comment (zerolab) [inline wagtail_airtable/templates/wagtail_airtable/airtable_import_listing.html:2]: suggestion: leave i18n for a separate PR, or make use of `trans`/`blocktrans` for all the strings

File: wagtail_airtable/views.py [hunk 3]
```diff
@@ -35,6 +36,10 @@ def post(self, request, *args, **kwargs):
         else:
             messages.add_message(request, messages.ERROR, "Could not import")
 
+        if (next_url := request.POST.get("next")) and url_has_allowed_host_and_scheme(
+            next_url, allowed_hosts=settings.ALLOWED_HOSTS, require_https=not settings.DEBUG
```
Comment (zerolab) [inline wagtail_airtable/views.py:40]: 👌🏼 for the `require_https` kwarg

having said that, any chance this can be covered by tests?

**Review by zerolab — COMMENTED** [review]
Did a quick drive-by review, with a few notes

File: pyproject.toml (hunk 1 above)
Comment (jams2) [reply pyproject.toml]: No reason not to, updated in https://github.com/wagtail/wagtail-airtable/pull/56/commits/850822907ad31e5f78a82ab83c4c3fc6653e9338

File: wagtail_airtable/templates/wagtail_airtable/airtable_import_listing.html (hunk 2 above)
Comment (jams2) [reply wagtail_airtable/templates/wagtail_airtable/airtable_import_listing.html]: Removed i18n in https://github.com/wagtail/wagtail-airtable/pull/56/commits/5836520a366014002972ba7d5f7ba4ac8f7c5621

File: wagtail_airtable/views.py (hunk 3 above)
Comment (jams2) [reply wagtail_airtable/views.py:40]: Tests added in https://github.com/wagtail/wagtail-airtable/pull/56/commits/92b960d6b1d84bea33429741236cd92a27f6ef63

**Review by zerolab — APPROVED** [review]
The changes look good to me.
Caveat: I don't have any build using this, so did not test locally

**Comment by gasman** [comment]
FYI, I've opened https://github.com/wagtail/wagtail/issues/11275 to hopefully make the template override less brittle in future :-)

**Review by gasman — APPROVED** [review]
Tested and working nicely on Wagtail 5.2 / Django 5.0!
